### PR TITLE
Add error message checking into existing python tests

### PR DESCRIPTION
**Pull request by stiepan**
Signed-off-by: Kamil Tokarski <ktokarski@nvidia.com>

Add check against expected messages to following python assert_raises/raises tests:
dali/test/python/test_operator_squeeze.py
dali/test/python/test_dali_tf_dataset_shape.py
dali/test/python/test_external_source_impl_utils.py
dali/test/python/test_external_source_parallel.py
dali/test/python/test_operator_crop_mirror_normalize.py
dali/test/python/test_external_source_impl.py
dali/test/python/test_operator_random_object_bbox.py

## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [x] **Other** (*e.g. Documentation, Tests, Configuration*)

#### Additional information
- Affected modules and functionalities:

- Key points relevant for the review:

### Tests
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2234

### CODE REVIEW

File: dali/test/python/test_operator_random_object_bbox.py [hunk 1]
```diff
@@ -457,27 +458,28 @@ def _test_err_args(**kwargs):
...
 def test_err_classes_bg():
-    with assert_raises(RuntimeError):
+    with assert_raises(RuntimeError, glob="Class label 0 coincides with background label"):
         _test_err_args(classes=[0,1,2,3], background=0)
 
 def test_err_classes_weights_length_clash():
-    with assert_raises(RuntimeError):
-        _test_err_args(classes=[0,1,2,3], weights=np.float32([1,2,3]))
-    with assert_raises(RuntimeError):
-        _test_err_args(classes=np.int32([0,1,2,3]), weights=[3,2,1])
```
Comment (stiepan) [inline dali/test/python/test_operator_random_object_bbox.py:467]: Renamed kwargs 
```
weights -> class_weights
ignore_classes -> ignore_class
```
to meet api names, I believe that was intended check here.

**Comment by stiepan** [comment]
!build

File: dali/test/python/test_dali_tf_dataset_shape.py [hunk 2]
```diff
@@ -309,11 +301,13 @@ def test_deprecated():
...
+    yield raises(ValueError, error_msg.format(*(("dtypes",) * 3)))(dali_pipe_deprecated), \
             { "shapes": 2, "dtypes": tf.uint8, "output_dtypes": tf.uint8 }, 2, tf.uint8, dali_types.UINT8, 1, 2
 
 def test_no_output_dtypes():
-    yield dali_pipe_deprecated_raises, \
-            { "shapes": 2, }, 2, tf.uint8, dali_types.UINT8, 1, 2
\ No newline at end of file
+    yield raises(TypeError,
+        "`output_dtypes` should be provided as single tf.DType value or a tuple of tf.DType values")(
+            dali_pipe_deprecated), { "shapes": 2, }, 2, tf.uint8, dali_types.UINT8, 1, 2
```
Comment (klecki) [inline dali/test/python/test_dali_tf_dataset_shape.py:313]: If any fixes appear, pleas add the missing newline

File: dali/test/python/test_dali_tf_dataset_shape.py [hunk 3]
```diff
@@ -309,11 +301,13 @@ def test_deprecated():
 
 
 def test_deprecated_double_def():
-    yield dali_pipe_deprecated_raises, \
-            { "shapes": 2, "output_shapes": 2, "dtypes": tf.uint8 }, 2, tf.uint8, dali_types.UINT8, 1, 2
-    yield dali_pipe_deprecated_raises, \
+    error_msg = "Usage of `{}` is deprecated in favor of `output_{}`*only `output_{}` should be provided."
+    yield raises(ValueError, error_msg.format(*(("shapes",) * 3)))(dali_pipe_deprecated), \
```
Comment (klecki) [inline dali/test/python/test_dali_tf_dataset_shape.py:305]: maybe 
```
with assert_raises(ValueError, glob=error_msg.format(...)):
   ...
```
would be less of a parentheses rollecoaster?

**Review by klecki — APPROVED** [review]
Just some nitpicks.

**Review by szalpal — APPROVED** [review]

File: dali/test/python/test_external_source_parallel.py [hunk 4]
```diff
@@ -52,15 +53,21 @@ def generator_fun():
...
 
 def test_wrong_source():
-    for source in disallowed_sources:
-        yield check_source_build, source
+    common_msg = "External Source in parallel mode (when `parallel=True`) accepts as `source` only *. Got {} instead"
+    expected_error_msgs = (
+        common_msg.format("a callable that does not accept arguments"),
+        "External source callback must be a callable with 0 or 1 argument",
+        common_msg.format("an iterable"),
+        common_msg.format("a generator function"))
```
[the remark this answers is not in the document]
Comment (stiepan) [reply dali/test/python/test_external_source_parallel.py:67]: Not really, subconscious preference I guess.

[the remark this answers is not in the document]
Comment (stiepan) [reply dali/test/python/test_external_source_parallel.py]: done

File: dali/test/python/test_operator_squeeze.py [hunk 5]
```diff
@@ -56,17 +56,30 @@ def test_squeeze():
...
         ([2], "Z", "XYZ", [[1, 1, 10]]),
         ([2, 1], None, "XYZ", [(100, 0, 0)]),
         ([1, 1], None, "XYZ", [(300, 1, 200), (10, 1, 10)]),
     ]
-    for axes, axis_names, layout, shapes in args:
-        pipe = squeeze_pipe(batch_size=len(shapes), num_threads=1, device_id=0, shapes=shapes, axes=axes, axis_names=axis_names, layout=layout)
-        with assert_raises(RuntimeError):
-            pipe.build()
-            pipe.run()
+    expected_errors = (
```
[the remark this answers is not in the document]
Comment (stiepan) [reply dali/test/python/test_operator_squeeze.py]: done

File: dali/test/python/test_dali_tf_dataset_shape.py (hunk 2 above)
Comment (stiepan) [reply dali/test/python/test_dali_tf_dataset_shape.py]: done

File: dali/test/python/test_dali_tf_dataset_shape.py (hunk 3 above)
Comment (stiepan) [reply dali/test/python/test_dali_tf_dataset_shape.py]: added named variables for error message strings

**Comment by stiepan** [comment]
!build

**Pull request merged**